protoutil/blockutils.go

```diff
@@ -156,16 +156,30 @@ func GetConsenterMetadataFromBlock(block *cb.Block) (*cb.Metadata, error) {
 // GetLastConfigIndexFromBlock retrieves the index of the last config block as
 // encoded in the block metadata
 func GetLastConfigIndexFromBlock(block *cb.Block) (uint64, error) {
-	md, err := GetMetadataFromBlock(block, cb.BlockMetadataIndex_LAST_CONFIG)
+	m, err := GetMetadataFromBlock(block, cb.BlockMetadataIndex_SIGNATURES)
 	if err != nil {
-		return 0, err
+		return 0, errors.Wrap(err, "failed to get signatures")
 	}
-	lc := &cb.LastConfig{}
-	err = proto.Unmarshal(md.Value, lc)
+	// TODO FAB-15864 Remove this fallback when we can stop supporting upgrade from pre-1.4.1 orderer
+	if len(m.Value) == 0 {
+		m, err := GetMetadataFromBlock(block, cb.BlockMetadataIndex_LAST_CONFIG)
```

Comment:
Potential for out of bounds dereference.  I'd think it would be more appropriate to move the len checks in here rather than always sprinkle them in front of this call.